src/vuegraf/vuegraf.py

```diff
@@ -119,24 +119,41 @@ def extractDataPoints(device, usageDataPoints):
     for chanNum, chan in device.channels.items():
         if chan.nested_devices:
             for gid, nestedDevice in chan.nested_devices.items():
-                extractDataPoints(nestedDevice, usageDataPoints)
+                extractDataPoints(nestedDevice, usageDataPoints, historyStartTime, historyEndTime)
+
+        chanName = lookupChannelName(account, chan)
 
         kwhUsage = chan.usage
         if kwhUsage is not None:
-            chanName = lookupChannelName(account, chan)
             watts = float(minutesInAnHour * wattsInAKw) * kwhUsage
             timestamp = stopTime
```

Comment:
@jertel This `stopTime` is not explicitly passed into the function. That could be implemented on a future PR.

Reply:
Yes there are multiple cases of this. It started out as a few lines of python scripting but feature requests have caused it to become more complex. A refactor is probably due at some point.